packages/foam-vscode/src/features/folder-rename-handler.ts

```diff
@@ -0,0 +1,1112 @@
+          let actualSourceUri = sourceFileUri;
+          const oldFolderPath = oldFolderUri.fsPath.replace(/\\/g, '/');
+          const sourceFilePath = sourceFileUri.fsPath.replace(/\\/g, '/');
+
+          if (sourceFilePath.startsWith(oldFolderPath)) {
+            // Source file is also in the renamed folder, so calculate its new path
+            actualSourceUri = this.calculateNewFileUri(
+              sourceFileUri,
+              oldFolderUri,
+              newFolderUri
+            );
+            Logger.debug(
+              `  → Source file also moved to: ${vscode.workspace.asRelativePath(
+                actualSourceUri
+              )}`
+            );
+          }
+
+          switch (connection.link.type) {
+            case 'wikilink': {
+              const targetFoamUri = fromVsCodeUri(newFileUri); // URI of the moved file (link target)
+              const { target: originalTarget } = MarkdownLink.analyzeLink(
+                connection.link
+              );
+
+              // Check if the original wikilink identifier would still resolve correctly after the rename
+              // If it's a simple identifier (no path separators), and it would still be unique, preserve it
+              let shouldPreserveOriginal = false;
+
+              // Only consider preserving simple identifiers (no path separators)
+              // Path-based identifiers like "features/daily-notes" need to be updated when folders are renamed
+              if (
+                !originalTarget.includes('/') &&
+                FoamWorkspace.isIdentifier(originalTarget)
+              ) {
+                // This is a simple identifier like [[note]] or [[daily-notes]]
+                // Check if it would still resolve uniquely to our target after the rename
+
+                // Get all resources that match this identifier (workspace is already updated)
+                const candidatesAfterRename =
+                  this.foam.workspace.listByIdentifier(originalTarget);
+
+                // If there's exactly one candidate and it's our target, the simple identifier remains unambiguous
+                if (
+                  candidatesAfterRename.length === 1 &&
+                  candidatesAfterRename[0].uri.isEqual(targetFoamUri)
+                ) {
+                  shouldPreserveOriginal = true;
+                  Logger.debug(
+                    `  → Preserving simple wikilink "${originalTarget}" as it remains unambiguous after rename`
+                  );
+                } else {
+                  Logger.debug(
+                    `  → Simple wikilink "${originalTarget}" has ${candidatesAfterRename.length} candidates, will update to avoid ambiguity`
+                  );
+                }
+              } else {
+                Logger.debug(
+                  `  → Path-based wikilink "${originalTarget}" will be updated for folder rename`
+                );
```

Comment:
I am not sure I understand this section. If it's a simple wikilink, renaming the folder shouldn't affect the link (it was `[[filename]]` before and it will be `[[filename]]` after.
In the case where there is more than one candidate, I don't think you have enough information to know which candidate should be kept, which I guess means we would still want to leave the original wikilink.
Am I missing or misunderstanding something?